src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -784,42 +785,69 @@
 
 (defn-validated delete-pending-path-id-orphans!
   "Delete paths in dropped-pids that are no longer mentioned
-  in other factsets."
+   in other factsets."
   [factset-id dropped-pids]
   (when-let [dropped-pids (seq dropped-pids)]
     (let [in-pids (jdbc/in-clause dropped-pids)]
-      (sql/do-prepared
-       (format
-        "DELETE FROM fact_paths fp
-           WHERE fp.id %s
-             AND NOT EXISTS (SELECT 1 FROM facts f
-                               WHERE f.fact_path_id %s
-                                 AND f.fact_path_id = fp.id
-                                 AND f.factset_id <> ?)"
-        in-pids in-pids)
-       (concat dropped-pids dropped-pids [factset-id])))))
+      (if (sutils/postgres?)
+        (sql/do-prepared
+          "DELETE FROM fact_paths fp
+           WHERE fp.id = ANY (?)
+           AND NOT EXISTS (SELECT 1 FROM facts f
+             WHERE f.fact_path_id = ANY (?)
+             AND f.fact_path_id = fp.id
+             AND f.factset_id <> ?)"
+          (flatten [(sutils/array-to-param "bigint" Long dropped-pids)
```

Comment:
Does this have to be two calls to array-to-param, or can we pass the same object in both positions? Also, how does `flatten` work here? That doesn't make sense to me.